src/backend/distributed/commands/table.c

```diff
@@ -391,11 +391,18 @@ PostprocessCreateTableStmtPartitionOf(CreateStmt *createStatement, const
 			return;
 		}
 
+		char *parentRelationName = generate_qualified_relation_name(parentRelationId);
+
+		if (IsCitusTableType(parentRelationId, NULL_KEY_DISTRIBUTED_TABLE))
+		{
+			CreateNullShardKeyDistTable(relationId, parentRelationName);
+			return;
+		}
```

Comment:
I was wondering if we should do that after SwitchToSequentialAndLocalExecutionIfPartitionNameTooLong, but I guess this is not needed because we're definitely creating a single shard, right?

Should we maybe add a one-liner comment for that?

Reply:
(If that makes sense, we can add the same comment for attach partition too)